### DISCUSSION TITLE
Threading error when using python-socketio for a few mintutes.

### DISCUSSION THREAD

**Discussion by tristan-f-r** [miguelgrinberg/python-socketio#929] [Q&A] [answered]
**Describe the bug**
Whenever I use python-socketio on a raspberry PI, eventually after a few minutes sending a lot of packets causes a threading error which crashes the bot, which is unacceptable when working with robots.

**Exception**
```
Exception in thread Thread-3:
Traceback (most recent call last)
  File "/usr/lib/python3.7/threading.py...
  engineio/client.py, line 367

RuntimeError: can't start new thread

packet queue is empty, aborting
```

Do note, that this program is sending max 30 packets every second.

**Comment by miguelgrinberg** [comment, marked as answer]
This has nothing to do with Socket.IO. You are starting too many threads, or you are leaking threads in your application.
